### PR TITLE
fix: Vue simulator renders the embed-only interface

### LINKED ISSUE #1078: 🐞 Bug: [vuesim] Circuit preview loads full simulator instead of embed version
### Is there an existing issue for this?

- [x] I have searched the existing issues.

### Describe the bug

The preview of the circuit shows the entire simulator version and not the embed vue one. 

### Steps to Reproduce

1. Go to the main devise website (`localhost:3000`)
2. Login as admin
3. go to flipper and enable vuesim (`localhost:3000/flipper`)
4. go to `localhost:3000/simulator` and create a circuit
5. save the project online with a name
6. when prompted, open the project page in the main website
7. can see that the preview shown is not the embed version, but the entire simulator features are visible

### Expected Behavior

Image

the embed version has limited features available and not the entire simualator

### Screenshots

https://github.com/user-attachments/assets/3720205d-aaa3-4f23-821a-e397620c4714

### Is the faced issue/bug related to the Vue simulator?

Yes

### Vue simulator version

v0, v1

### Used Vue simulator with or without backend?

With Backend

### Is the bug present only on the dev server, the build, or both?

Both

### Device Information

```markdown

```

### Additional Context

part of gsoc'26 (prj-4)

### Are you working on this issue?

Yes

### PR DESCRIPTION
**Pull request by ShinichiShi**
Fixes #1078 

 

#### Describe the changes you have made in this PR -
This PR fixes an issue where the Vue simulator embed page rendered the full simulator UI instead of the intended embed-only interface.

Root Cause: `App.vue` determines whether to render the embed simulator using:
   `const isEmbed = computed(() => (window as any).embed === true)`

The Rails embed template correctly initializes: `window.embed = true`. However, `globalVariables.ts` later overwrote this value during module execution: `window.embed = false;`

Since ES module scripts are deferred by default, the execution order became:
Rails inline script sets window.embed = true -> Vue module bundle loads -> `globalVariables.ts` resets it to false -> Vue initializes in non-embed mode

As a result, the project iframe displayed the full simulator instead of the restricted embed UI.

### Screenshots of the UI changes (If any) -

No ui changes
image

---

## Code Understanding and AI Usage

**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**
- [x] No, I wrote all the code myself
- [ ] Yes, I used AI assistance (continue below)

**If you used AI assistance:**
- [ ] I have reviewed every single line of the AI-generated code
- [ ] I can explain the purpose and logic of each function/component I added
- [ ] I have tested edge cases and understand how the code handles them
- [ ] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**

Updated the initialization logic in globalVariables.ts:  `window.embed = window.embed ?? false;` 
This preserves any value already assigned by the Rails template while still providing a default value for the standalone simulator flow.

I considered a few alternatives:
- Moving the embed assignment into another module script
- Passing embed mode through meta tags
- Removing the initialization entirely
- Using build-time environment variables

The approach i used was minimalistic change without any major fix

---

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [ ] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

---

This PR will be followed by another PR in the main repo, where the circuit is rendered within the iframe. Currently there's an issue with the sanitize data and needs to be fixed

Note: Please check **Allow edits from maintainers** if you would like us to assist in the PR.

## Summary by CodeRabbit

* **Refactor**
  * Improved initialization logic to preserve pre-existing global configuration values and login state. The application now maintains previously set values instead of unconditionally resetting them to defaults during startup.

[Review Change Stack](https://app.coderabbit.ai/change-stack/CircuitVerse/cv-frontend-vue/pull/1082?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: v1/src/globalVariables.ts [hunk 1]
```diff
@@ -6,8 +6,8 @@ declare const window: any;
 import jQuery from "jquery";
 window.$ = window.jQuery = jQuery;
 
-window.isUserLoggedIn = false;
-window.logixProjectId = undefined;
+window.isUserLoggedIn = window.isUserLoggedIn ?? false;
+window.logixProjectId = window.logixProjectId ?? undefined;
```
[the remark this answers is not in the document]
Comment (ShinichiShi) [reply v1/src/globalVariables.ts]: fixed

**Comment by tachyons** [comment]
@ShinichiShi Could you fix the linting errors ?
